ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -126,6 +128,12 @@ public AmbryRequests(StorageManager storageManager, RequestResponseChannel reque
     requestsDisableInfo.put(RequestOrResponseType.DeleteRequest, Collections.newSetFromMap(new ConcurrentHashMap<>()));
     requestsDisableInfo.put(RequestOrResponseType.ReplicaMetadataRequest,
         Collections.newSetFromMap(new ConcurrentHashMap<>()));
+
+    Set<PartitionId> partitionIds = new HashSet<>();
+    for (ReplicaId replicaId : clusterMap.getReplicaIds(currentNode)) {
+      partitionIds.add(replicaId.getPartitionId());
+    }
+    partitionsInCurrentNode = Collections.unmodifiableList(new ArrayList<>(partitionIds));
```

Comment:
Is it possible for this to remain a `Collection`/`Set` instead of being copied into a `List`? I think there are a few methods in this class that take in the `List` but it seems like they just rely on it being iterable (I could have missed something, though)

Reply:
Good suggestion. However, since "partitionsOnNode" is small and constant size, I don't think list or set (or any other structure) makes a difference. However, it was easy enough to convert this into `Collection` and since that looks like the right thing to do (`List` is not necessary), I have made the changes.